app/controllers/lectures_controller.rb

```diff
@@ -116,8 +116,18 @@ def end_lecture
 
   private
     # Use callbacks to share common setup or constraints between actions.
+    def get_course
+      @course = Course.find_by(id: params[:course_id])
+      if @course.nil?
+        redirect_to root_path, alert: "The course you requested does not exist."
+      end
+    end
+
     def set_lecture
```

Comment:
Is this a get or a set? It is somehow confusing, maybe write a comment and call it `get_lecture`.